### ISSUE TITLE
Cannot use ResourceSaver and ResourceLoader sequentially for PNG files

### ISSUE THREAD

**Issue by rainlizard** [godotengine/godot#55226]
### Godot version

3.4 stable

### System information

Windows 8.1

### Issue description

You might wonder why someone would want to do this, but it's just two separate parts of my code that I stripped away the rest from to find the issue. I think this is still unexpected behaviour regardless. "Returns OK on success" is surely a lie, why is it calling it a success if it's not even waiting until the save is fully complete?

When saving .RES files the method waits until the save is fully complete but it doesn't wait when saving .PNG, there's an inconsistency there.

### Steps to reproduce

You can reproduce the issue by using ``ResourceSaver.save()`` then ``load()`` (or ResourceLoader) and the loaded file will be printed as ``null``.

```
func _ready():
	var imgTex = ImageTexture.new()
	imgTex.create_from_image(load("res://icon.png").get_data())
	
	ResourceSaver.save("res://unloadable.png", imgTex)
	var resource = load("res://unloadable.png")
	print(resource)
	
	Directory.new().remove("res://unloadable.png") # just cleaning up for the next time you run this
```

### Minimal reproduction project

[ResourceSaver.zip](https://github.com/godotengine/godot/files/7580923/ResourceSaver.zip)

**Comment by timothyqiu** [comment]
To `load()` a resource, you have to import it in the editor first. The newly saved `res://unloadable.png` has never been imported in the editor, so the following `load()` will definitely fail, by design.

If you comment out the last `remove()` line and run the project from editor for two times:

* On the first run, it will generate the png and fail to load.
* Then the project exits and bring up the editor, the editor will import the newly created png automatically.
* On the second run, the `load()` will successfully return a texture.

**Comment by rainlizard** [comment]
Oops I thought load() was working. The idea is to load an external file from a person's hard drive, not an internal godot file.

The only other way I can find seems to be to use ``Image`` class's load(), however that seems inefficient as it then needs to be converted to a texture every time. What if someone wanted to store all their game images externally? They have to convert all of them to textures.

Except in the case of .res which works as I said:
```
ResourceSaver.save("res://unloadable.res", imgTex)
var resource = load("res://unloadable.res")
print(resource)

Directory.new().remove("res://unloadable.res") # cleaning up for the next time you run this
```
But I guess res just saves extra data so the importer isn't required.

Anyway I thought for sure I spotted an inconsistency, a bug, but maybe they've just got different rules and this is expected. Closing.
